### PR TITLE
github: add new pr requirement for linking issue and commit message

### LINKED ISSUE #11886: update PR template for linking issue and commit message
see tikv/community#162

### PR DESCRIPTION
**Pull request by zhangyangyu**
Issue Number: close #11886 

### Release notes

```release-note
None
```

### CODE REVIEW

**Comment by zhangyangyu** [comment]
@Mini256 @BusyJay @sunxiaoguang PTAL

**Comment by Mini256** [comment]
I think in most cases, the contents of the  "what's change" and "commit message" parts will be repeated. Should we separate them? 🤔

**Comment by zhangyangyu** [comment]
> I think in most cases, the contents of the "what's change" and "commit message" parts will be repeated. Should we separate them? 🤔

Not sure. IMHO a good commit message's body describes **what** and **why**. Not sure contributors would treat *what's changed* to include all of them.

https://cbea.ms/git-commit/#why-not-how

**Comment by Mini256** [comment]
\cc @tikv/maintainers ping~

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -38,6 +47,17 @@ Side effects
     - Consumes more MEM
 - Breaking backward compatibility
 
+### Commit message
```
Comment (BusyJay) [inline .github/pull_request_template.md:50]: How about merging it with "What's change"?

**Review by BusyJay — COMMENTED** [review]
Rest LGTM

**Review by BusyJay — APPROVED** [review]

**Comment by sunxiaoguang** [comment]
/run-idc_jenkins_ci/test

**Comment by sunxiaoguang** [comment]
/run-idc_jenkins_ci/test

**Review by sunxiaoguang — APPROVED** [review]
LGTM

**Pull request merged**